wagtail/core/management/commands/replace_text.py

```diff
@@ -32,9 +35,12 @@ def handle(self, *args, **options):
         from_text = options["from_text"]
         to_text = options["to_text"]
 
-        for revision in PageRevision.objects.filter(content_json__contains=from_text):
-            revision.content_json = revision.content_json.replace(from_text, to_text)
-            revision.save(update_fields=["content_json"])
+        for revision in PageRevision.objects.annotate(
+            content_text=Cast("content", output_field=models.TextField())
+        ).filter(content_text__contains=from_text):
+            replacement = revision.content_text.replace(from_text, to_text)
+            revision.content = json.loads(replacement)
+            revision.save(update_fields=["content"])
```

Comment:
This isn't ideal because we treat the JSON as a string. However, this command dates all the way back to https://github.com/wagtail/wagtail/blob/5c69d3ae23e5389c70685bc181d21890c449a3ea/wagtail/wagtailcore/management/commands/replace_text.py

We probably should deprecate this.

Reply:
I agree, I'll raise it at the core team meeting tomorrow.